operator/pkg/values/translation/translation.go

```diff
@@ -138,40 +138,88 @@ func GetGlobalMonitoring(m *platform.GlobalMonitoring) *ValuesBuilder {
 		}
-		autoscaling.SetInt32("minReplicas", scaling.MinReplicas)
-		autoscaling.SetInt32("maxReplicas", scaling.MaxReplicas)
-		sv.AddChild("autoscaling", &autoscaling)
 	}
+}
 
-	if analyzer != nil {
-		sv.SetStringMap("nodeSelector", analyzer.NodeSelector)
-		sv.AddChild(ResourcesKey, GetResources(analyzer.Resources))
-		sv.AddAllFrom(GetTolerations(TolerationsKey, analyzer.DeploymentSpec.Tolerations))
+// SetScannerAnalyzerValues sets values in "sv" based on "analyzer".
+func SetScannerAnalyzerValues(sv *ValuesBuilder, analyzer *platform.ScannerAnalyzerComponent) {
+	if analyzer == nil {
+		return
 	}
+	setScannerComponentScaling(sv, analyzer.GetScaling())
+	sv.SetStringMap("nodeSelector", analyzer.NodeSelector)
+	sv.AddChild(ResourcesKey, GetResources(analyzer.Resources))
+	sv.AddAllFrom(GetTolerations(TolerationsKey, analyzer.DeploymentSpec.Tolerations))
 }
 
-// SetScannerDBValues sets values in "sb" based on "db".
+// SetScannerDBValues sets values in "sv" based on "db".
 func SetScannerDBValues(sv *ValuesBuilder, db *platform.DeploymentSpec) {
 	if db != nil {
 		sv.SetStringMap("dbNodeSelector", db.NodeSelector)
 		sv.AddChild("dbResources", GetResources(db.Resources))
 		sv.AddAllFrom(GetTolerations("dbTolerations", db.Tolerations))
 	}
 }
+
+// SetScannerV4DBValues sets values in "sv" based on "db"
+func SetScannerV4DBValues(sv *ValuesBuilder, db *platform.ScannerV4DB) {
+	if db == nil {
+		return
+	}
+
+	dbVB := NewValuesBuilder()
+	dbVB.SetStringMap("nodeSelector", db.NodeSelector)
+	dbVB.AddChild(ResourcesKey, GetResources(db.Resources))
+	dbVB.AddAllFrom(GetTolerations(TolerationsKey, db.Tolerations))
+	// TODO(ROX-19051): translate persistence values
+	sv.AddChild("db", &dbVB)
+	return
+}
+
+// SetScannerV4ComponentValues sets values in "sv" based on "component"
+func SetScannerV4ComponentValues(sv *ValuesBuilder, componentKey string, component *platform.ScannerV4Component) {
+	if component == nil {
+		return
+	}
+
+	componentVB := NewValuesBuilder()
+	setScannerComponentScaling(sv, component.Scaling)
+	sv.SetStringMap("nodeSelector", component.NodeSelector)
+	sv.AddChild(ResourcesKey, GetResources(component.Resources))
+	sv.AddAllFrom(GetTolerations(TolerationsKey, component.Tolerations))
+	sv.AddChild(componentKey, &componentVB)
+}
+
+func setScannerComponentScaling(sv *ValuesBuilder, scaling *platform.ScannerComponentScaling) {
+	if scaling == nil {
+		return
+	}
+
+	sv.SetInt32("replicas", scaling.Replicas)
+	autoscalingVB := NewValuesBuilder()
+	if scaling.AutoScaling != nil {
+		switch *scaling.AutoScaling {
+		case platform.ScannerAutoScalingDisabled:
+			autoscalingVB.SetBoolValue("disable", true)
+		case platform.ScannerAutoScalingEnabled:
+			autoscalingVB.SetBoolValue("disable", false)
+		default:
+			autoscalingVB.SetError(fmt.Errorf("invalid scanner autoscaling %q", *scaling.AutoScaling))
+		}
+	}
+
+	autoscalingVB.SetInt32("maxReplicas", scaling.MaxReplicas)
```

Comment:
Or for example max < min. Don't know what would happen in such situations -- would it deploy and fail silently, for example.

Reply:
Good questions, currently I don't think this is validated. With 1.25 a feature was graduated to beta to support that in CRDs, I think recent controller-gen version also support the markers in Go code.

IMHO it would also be ok to leave it as is, since this is the same logic as old scanner and I've never heard any complains about that.
 https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules